### PR TITLE
Add SSL session timeout and cache size server configs

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
This change adds the SSL session timeout and session cache size config
parameters to the HTTP server.

### CODE REVIEW

File: http-server/src/main/java/io/airlift/http/server/HttpServerConfig.java [hunk 1]
```diff
@@ -70,6 +70,11 @@
     private List<String> includedCipherSuites = ImmutableList.of();
     private List<String> excludedCipherSuites = ImmutableList.of();
 
+    // For the SSL session-related config parameters
+    // a value of -1 means using the JVM default
+    private int sslSessionTimeout = -1;
```
Comment (dain) [inline http-server/src/main/java/io/airlift/http/server/HttpServerConfig.java]: This should be a duration

Comment (dain) [reply http-server/src/main/java/io/airlift/http/server/HttpServerConfig.java]: Oh also, lets have defaults.... 10k and 4 hours is most likely a fine place to start.

**Comment by nezihyigitbasi** [comment]
@dain addressed the comments.

File: http-server/src/main/java/io/airlift/http/server/HttpServerConfig.java [hunk 2]
```diff
@@ -162,6 +166,30 @@ public HttpServerConfig setHttpsPort(int httpsPort)
         return this;
     }
 
+    public Duration getSslSessionTimeout()
+    {
+        return sslSessionTimeout;
+    }
+
+    @Config("http-server.https.ssl-session-timeout")
+    public HttpServerConfig setSslSessionTimeout(Duration sslSessionTimeout)
```
Comment (dain) [inline http-server/src/main/java/io/airlift/http/server/HttpServerConfig.java]: add min duration of second

**Comment by nezihyigitbasi** [comment]
That test doesn't use https and I don't see how these changes would cause it. But, it's weird that it fails. I tried reproducing it locally with 1K runs, but cannot.

**Comment by nezihyigitbasi** [comment]
I added a commit with some logging to see what's going on.

**Comment by dain** [comment]
And then it passed :(

**Comment by nezihyigitbasi** [comment]
That test is totally flaky. I tried many times on Travis, but couldn't repro so far.

**Comment by dain** [comment]
We can fix that in another PR.  Can you remove that commit and I'll land this.

**Comment by nezihyigitbasi** [comment]
Removed.

**Pull request merged**